services/src/main/java/org/keycloak/services/resources/KeycloakApplication.java

```diff
@@ -145,29 +151,35 @@ public void run(KeycloakSession session) {
                 // TODO up here ^^
 
                 ApplianceBootstrap applianceBootstrap = new ApplianceBootstrap(session);
-                exportImportManager[0] = new ExportImportManager(session);
-
-                boolean createMasterRealm = applianceBootstrap.isNewInstall();
-                if (exportImportManager[0].isRunImport() && exportImportManager[0].isImportMasterIncluded()) {
-                    createMasterRealm = false;
-                }
-
-                if (createMasterRealm) {
-                    applianceBootstrap.createMasterRealm();
+                var exportImportManager = bootstrapState.exportImportManager = new ExportImportManager(session);
+                bootstrapState.newInstall = applianceBootstrap.isNewInstall();
+                if (bootstrapState.newInstall) {
+                    boolean importingMaster = exportImportManager.isRunImport() && exportImportManager.isImportMasterIncluded();
+                    importingMaster |= getImportDirectory().filter(exportImportManager::isImportMasterIncludedAtStartup).isPresent();
```

Comment:
It'd be maybe good to add some comments for future generations explaining why we need to basically run `isImportMasterIncluded` twice. :)

Reply:
Ok, added some comments